flang/lib/Lower/IO.cpp

```diff
@@ -1120,7 +1120,7 @@ genBuffer(Fortran::lower::AbstractConverter &converter, mlir::Location loc,
   // Helper to query [BUFFER, LEN].
   Fortran::lower::CharacterExprHelper helper(builder, loc);
   auto dataLen = helper.materializeCharacterOrSequence(
-      fir::getBase(converter.genExprValue(*e)));
+      fir::getBase(converter.genExprAddr(*e)));
```

Comment:
I added a fix  here for a new fm909 lowering issue that appeared when using the runtime with the character array output fix.